devtools/gammacli/protocol.go

```diff
@@ -0,0 +1,60 @@
+package gammacli
+
+import (
+	"encoding/hex"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+)
+
+// Bytes are hex and are converted to []byte after json unmarshal
+type JSONMethodArgument struct {
+	Name  string
+	Type  string
+	Value interface{}
+}
+
+type JSONTransaction struct {
+	ContractName string
+	MethodName   string
+	Arguments    []JSONMethodArgument
+}
+
+type TransactionReceipt struct {
+	Txhash          primitives.Sha256
+	ExecutionResult protocol.ExecutionResult
+	OutputArguments []JSONMethodArgument
+}
+
+type SendTransactionOutput struct {
+	TransactionReceipt TransactionReceipt
+	TransactionStatus  protocol.TransactionStatus
+	BlockHeight        primitives.BlockHeight
+	BlockTimestamp     primitives.TimestampNano
+}
+
+type CallMethodOutput struct {
+	OutputArguments []JSONMethodArgument
+	CallResult      protocol.ExecutionResult
+	BlockHeight     primitives.BlockHeight
+	BlockTimestamp  primitives.TimestampNano
+}
+
+func (ma *JSONMethodArgument) String() string {
+	returnString := ma.Name + ":"
+
+	switch ma.Type {
+	case METHOD_ARGUMENT_TYPE_UINT32:
+		returnString = returnString + string(ma.Value.(uint32))
```

Comment:
I believe you can shorten this statement like this:

```
var value string

switch ma.Type {
       case METHOD_ARGUMENT_TYPE_UINT32:
       value = string(ma.Value.(uint32))
}

return ma.Name + ":" + value
```

Reply:
Ok.